[TIMOB-18099] Clean up tab events

AngelkPetkov opened the pull request:
JIRA: https://jira.appcelerator.org/browse/TIMOB-18099?filter=-1
Added code so selected/focus and blur/unselected can both be used to
call events.
Also added some debug code so the previous index doesn’t go to -1.

AngelkPetkov committed:
Merge remote-tracking branch 'appcelerator/master'

Added code so selected/focus and blur/unselected can both be used to
call events.
Also added some debug code so the previous index doesn’t go to -1.

cheekiatng commented on iphone/Classes/TiUITabGroup.m:
```diff
@@ -96,9 +96,21 @@ - (void)handleDidShowTab:(TiUITabProxy *)newFocus
-	NSInteger previousIndex = -1;
-	NSInteger index = -1;
+   
+    NSInteger previousIndex = 0;
```
check indentations. If this part of code is using tab, please use tab instead of 4 spaces. you can change this in the Xcode preferences.

cheekiatng commented:
Need to also document in apidoc, the yml files to:
include the newly introduced 'selected' and 'unselected'
include deprecated messages for focus / blur

AngelkPetkov committed:
[TIMOB]-18099

Line Formatting

AngelkPetkov commented:
The code's indentation is fine i don't know why its changing when i upload 
![screen shot 2015-09-18 at 10 03 29 am](https://cloud.githubusercontent.com/assets/14097072/9965688/aaa76396-5dec-11e5-8e78-547a2024dda4.png)

AngelkPetkov committed:
trying to fix index on github

AngelkPetkov committed:
[TIMOB]-18099

Warning message and updated .yml

cheekiatng commented on iphone/Classes/TiUITabGroup.m:
```diff
@@ -592,15 +619,23 @@ -(void)open:(id)args
 	[self addSubview:view];
 
-	// on an open, make sure we send the focus event to focused tab
+	// on an open, make sure we send the focu   s event to focused tab
```
fo    cus is messed up here.

AngelkPetkov committed:
[TIMOB-18099] Code indentation

AngelkPetkov committed:
[TIMOB -18099] comment

AngelkPetkov committed:
[TIMOB - 18099]  Spacing

AngelkPetkov committed:
[TIMOB]-18099 - Spacing

cheekiatng commented on iphone/Classes/TiUITabGroup.m:
```diff
@@ -115,7 +127,12 @@ - (void)handleDidShowTab:(TiUITabProxy *)newFocus
 
-	[self.proxy fireEvent:@"blur" withObject:event];
+	[self.proxy fireEvent:@"unselected" withObject:event];
+	if ([self.proxy _hasListeners:@"focus"]){
```
this should be _hasListeners:@"blur"

cheekiatng commented on iphone/Classes/TiUITabGroup.m:
```diff
@@ -115,7 +127,12 @@ - (void)handleDidShowTab:(TiUITabProxy *)newFocus
-	[self.proxy fireEvent:@"blur" withObject:event];
+	[self.proxy fireEvent:@"unselected" withObject:event];
+	if ([self.proxy _hasListeners:@"focus"]){
+		DEPRECATED_REPLACED(@"UI.TabGroup.Event:focus" ,@"5.1.0",@"UI.TabGroup.Event:selected")
```
this should be blur as well, and unselected

cheekiatng commented:
Please address new comments. also, note that the travis build failed, and if you see the Details, it has something to do with the yml file you editted. You may want to do this in the apidoc folder:
```
Go to the ./titanium_mobile/apidoc/ folder
Run "npm install ."
Run "node validate”
```
to find out where the problem is.

AngelkPetkov committed:
[TIMOB-18099] Spacing and namings

AngelkPetkov committed:
trying to fix yml

AngelkPetkov committed:
Should be fixed

AngelkPetkov committed:
don't even know any more

AngelkPetkov committed:
Should Fix yml spacing issue

AngelkPetkov committed:
Update GeolocationModule.m

AngelkPetkov committed:
Update app.js

AngelkPetkov committed:
Update app.js

AngelkPetkov committed:
Update TiUITabGroup.m

AngelkPetkov committed:
Update TabGroup.yml

AngelkPetkov committed:
Update TabGroup.yml

AngelkPetkov committed:
Update TiUITabGroup.m

AngelkPetkov committed:
Update TiUITabGroup.m

cheekiatng commented on iphone/Classes/TiUITabGroup.m:
```diff
@@ -599,8 +613,14 @@ -(void)open:(id)args
 	}
-	NSDictionary *event = [NSDictionary dictionaryWithObjectsAndKeys:focusedTabProxy,@"tab",NUMINTEGER(index),@"index",NUMINT(-1),@"previousIndex",[NSNull null],@"previousTab",nil];
-	[self.proxy fireEvent:@"focus" withObject:event];
+	NSDictionary *event = [NSDictionary dictionaryWithObjectsAndKeys:focusedTabProxy,@"tab",NUMINTEGER(index),@"index",[NSNull null],@"previousTab",nil];
```
previousIndex object is missing. is this deliberate?

cheekiatng commented:
Getting there! After you address the comment above, can you squash these commits please?

AngelkPetkov committed:
Accidental Code deletion restored

AngelkPetkov committed:
# This is a combination of 2 commits.
# The first commit's message is:

# This is a combination of 2 commits.
# The first commit's message is:

# This is a combination of 2 commits.
# The first commit's message is:

Trying to squash commits

# This is the 2nd commit message:

Update GeolocationModule.m

# This is the 2nd commit message:

Update app.js

# This is the 2nd commit message:

Update TabGroup.yml
